torch/csrc/autograd/variable.h

```diff
@@ -56,7 +56,10 @@ struct Variable : public at::Tensor {
   inline const std::shared_ptr<Function>& grad_fn() const;
   inline       std::shared_ptr<Function>& grad_fn();
 
+  // Sets the flags and grad_fn ("history") of a new Variable
   inline void set_history(VarFlags flags, int output_nr, std::shared_ptr<Function> grad_fn);
```

Comment:
I really thing this would have been much cleaner and safer if we just used the constructor for `set_history` since it's not new Variables only, and `rebase_history` for all other cases